### PR TITLE
While writing 126212 in Core does the same :)

### PR DESCRIPTION
**Pull request by CoMPaTech**
## Summary by CodeRabbit

- **New Features**
	- Updated the `biome` pre-commit hook to enhance linting capabilities for YAML files.
	- Introduced new linting rules for improved code quality checks.

- **Improvements**
	- Simplified device management process by renaming methods for synchronous operation.
	- Enhanced time manipulation in tests for better control and consistency.

- **Bug Fixes**
	- Restructured logic in device removal to improve accuracy and readability.

- **Chores**
	- Updated package dependencies in the testing environment.

### CODE REVIEW

**Comment by CoMPaTech** [comment]
@coderabbitai please add that this fixes https://github.com/plugwise/plugwise-beta/issues/717 and although writing myself at the time HA-Core PR 126212 fixed this even better

File: custom_components/plugwise/coordinator.py [hunk 1]
```diff
@@ -115,21 +115,21 @@ async def _async_update_data(self) -> PlugwiseData:
             raise ConfigEntryError("Device with unsupported firmware") from err
         else:
             LOGGER.debug(f"{self.api.smile_name} data: %s", data)
-            self._async_add_remove_devices(data, self.config_entry)
+            self._add_remove_devices(data, self.config_entry)
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply custom_components/plugwise/coordinator.py]: Good catch

File: custom_components/plugwise/coordinator.py [hunk 2]
```diff
@@ -138,16 +138,22 @@
...
         via_device: str = ""
+
+        # First find the Plugwise via_device
+        for device_entry in device_list:
+            for identifier in device_entry.identifiers:
+                if identifier[0] != DOMAIN or identifier[1] != data.gateway[GATEWAY_ID]:
+                    continue
+                via_device = device_entry.id
+                break
+
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply custom_components/plugwise/coordinator.py]: Added

File: tests/components/plugwise/test_init.py [hunk 3]
```diff
@@ -257,7 +257,8 @@ async def test_update_device(
     # Add a 2nd Tom/Floor
     data.devices.update(TOM)
     with patch(HA_PLUGWISE_SMILE_ASYNC_UPDATE, return_value=data):
-        async_fire_time_changed(hass, utcnow + timedelta(minutes=2))
+        freezer.tick(timedelta(minutes=10))
```
Comment (bouwew) [inline tests/components/plugwise/test_init.py:260]: minutes=1 should work, it's the default interval.

File: pyproject.toml [hunk 4]
```diff
@@ -33,7 +34,7 @@ target-version = ["py312"]
 exclude = 'generated'
 
 [tool.ruff]
-target-version = "py312"
+required-version = ">=0.5.3"
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply pyproject.toml:37]: Aligning with core

File: tests/components/plugwise/test_init.py (hunk 3 above)
Comment (CoMPaTech) [reply tests/components/plugwise/test_init.py:260]: Just down streaming :)

File: scripts/core-testing.sh [hunk 5]
```diff
@@ -31,7 +31,7 @@ echo "Working on HA-core branch ${core_branch}"
 # Which packages to install (to prevent installing all test requirements)
 # actual package version ARE verified (i.e. grepped) from requirements_test_all
 # separate packages with |
-pip_packages="fnvhash|lru-dict|voluptuous|aiohttp_cors|pyroute2|sqlalchemy|zeroconf|pytest-socket|pre-commit|paho-mqtt|numpy|pydantic|ruff|ffmpeg|hassil|home-assistant-intents"
+pip_packages="fnvhash|lru-dict|voluptuous|aiohasupervisor|aiohttp_cors|pyroute2|sqlalchemy|zeroconf|pytest-socket|pre-commit|paho-mqtt|numpy|pydantic|ruff|ffmpeg|hassil|home-assistant-intents"
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply scripts/core-testing.sh:34]: Yeah, nothing changed, so that's still ok

File: tests/components/plugwise/test_init.py (hunk 3 above)
Comment (bouwew) [reply tests/components/plugwise/test_init.py:260]: image

Comment (bouwew) [reply tests/components/plugwise/test_init.py:260]: Upstream staat dit er.

Comment (CoMPaTech) [reply tests/components/plugwise/test_init.py:260]: Fixed

**Review by bouwew — APPROVED** [review]
LGTM!

**Comment by bouwew** [comment]
Also, make this a v0.53.2 release?
This PR solves a potential failing device-removal. OK, not happening often, low risk, but still...

**Comment by CoMPaTech** [comment]
I was opting to fix the #717 too and go from there, not sure now why/how that is failing, lets release after that (and be ready for next release)

**Pull request merged**